src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -687,8 +687,22 @@ private async Task FetchGlobalAndScriptVariablesAsync()
             var scopeVariableContainer = new VariableContainerDetails(this.nextVariableId++, "Scope: " + scope);
             this.variables.Add(scopeVariableContainer);
 
-            IReadOnlyList<PSObject> results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
-                .ConfigureAwait(false);
+            IReadOnlyList<PSObject> results;
+            try
+            {
+                results = await _executionService.ExecutePSCommandAsync<PSObject>(psCommand, CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+            catch (CmdletInvocationException ex)
+            {
+                if (!(
+                    ex.ErrorRecord.CategoryInfo.Reason.Equals("PSArgumentOutOfRangeException") &&
+                    ex.Message.Contains("exceeds the number of active scopes")))
+                {
```

Comment:
Does this work with localization?

Reply:
Trying to see how @rjmholt would have done this...

Reply:
Ideally it just wouldn't throw here due to changes in `ExecutePSCommandAsync`, but if we do need the check here we could just look at `ex.ParamName is "Scope"`

Reply:
I'm reworking how the vars are processed right now so that this part will be removed, I can remove the message contains part, just wanted to get as specific as I could.

Reply:
I remember during code review Rob explained how it was possible to use `ExecutePSCommandAsync` (and friends) with errors either returned as exceptions or returned in the error stream...unfortunately I don't think we want the latter here so we are probably going to have to catch a specific exception. Or better yet (which it sounds like you're doing) not cause it to be generated in the first place.

Reply:
@andschwa that's something that happens based on what type of error it is considered to the PowerShell engine, terminating or non-terminating.  Terminating is surfaced by a throw, non-terminating goes through the error stream

Reply:
Narrowed it to just the reason and not the message, as this is very unlikely to happen now.